KVM v12.3.0 with helm3 changes (#480)
giantswarm/releases: kvm/v12.3.0/README.md

If you are still using Helm 2 then these Helm releases will not be affected. However we encourage you to upgrade to Helm 3. As Helm 2 support ends on November 13th 2020. https://helm.sh/blog/helm-v2-deprecation-timeline/

Below, you can find more details on components that were changed with this release.

**Note before upgrade:**

Please contact your Solution Engineer before upgrading. The upgrade is automated. However, it includes a data migration from Helm 2 release configmaps to Helm 3 release secrets, there are some pre-upgrade checks and we recommend monitoring the upgrade to ensure safety.

**Note for Solution Engineers:**

Please use [Upgrading tenant clusters to Helm 3](https://intranet.giantswarm.io/docs/dev-and-releng/helm/helm3-tenant-cluster-upgrade/) as a guide on the upgrade process for the checks and monitoring steps.

**Note for future 12.x.x releases:**

Please persist this note and the above, until all customers are on AWS v12.2.x and above.

## Change details

> ### app-operator [v2.3.1](https://github.com/giantswarm/app-operator/blob/master/CHANGELOG.md#231---2020-09-22)

Reviewer: ```suggestion
### app-operator [v2.3.3](https://github.com/giantswarm/app-operator/blob/master/CHANGELOG.md#233---2020-10-15)
```

Author: thanks for finding it!

## :zap:  Giant Swarm Release v12.3.0 for KVM :zap:

This release upgrades all Helm releases managed by Giant Swarm to use [Helm v3.3.4](https://github.com/helm/helm/releases/tag/v3.3.4).

This lets us benefit from the improved security model and keep up to date with the community. We also remove the Tiller deployment from the giantswarm namespace, removing its gRPC endpoint, which reduces operational complexity.

If you are still using Helm 2 then these Helm releases will not be affected. However we encourage you to upgrade to Helm 3. As Helm 2 support ends on November 13th 2020. https://helm.sh/blog/helm-v2-deprecation-timeline/

> 

Reviewer 2: ```suggestion

The release also upgrades Container Linux to address security vulnerabilities.

```

Below, you can find more details on components that were changed with this release.

**Note before upgrade:**

Please contact your Solution Engineer before upgrading. The upgrade is automated. However, it includes a data migration from Helm 2 release configmaps to Helm 3 release secrets, there are some pre-upgrade checks and we recommend monitoring the upgrade to ensure safety.

**Note for Solution Engineers:**

> Please use [Upgrading tenant clusters to Helm 3](https://intranet.giantswarm.io/docs/dev-and-releng/helm/helm3-tenant-cluster-upgrade/) as a guide on the upgrade process for the checks and monitoring steps.

Reviewer 2: ```suggestion
Please ensure cluster is on [KVM 12.1.x](https://github.com/giantswarm/releases/tree/master/kvm/v12.1.0) platform release first before upgrading to 12.2.0+

Please use [Upgrading tenant clusters to Helm 3](https://intranet.giantswarm.io/docs/dev-and-releng/helm/helm3-tenant-cluster-upgrade/) as a guide on the upgrade process for the checks and monitoring steps.
```

**Note for future 12.x.x releases:**

> Please persist this note and the above, until all customers are on AWS v12.2.x and above.

Reviewer 2: ```suggestion
Please ensure cluster is on [KVM 12.1.x](https://github.com/giantswarm/releases/tree/master/kvm/v12.1.0) platform release first before upgrading to 12.2.0+ Please persist this note until all customers are on KVM 12.1.x and above.
```
@giantswarm/app-squad-nginx , can you check this note is appropriate? This is related to making nginx optional in kvm 12.1.0

Reviewer 3: Adding that note makes sense, but I'm not sure it's on the right place.

```
Please ensure cluster is on [KVM 12.1.x](https://github.com/giantswarm/releases/tree/master/kvm/v12.1.0) platform release first before upgrading to 12.2.0+
```
could be part of the "Note for SEs", and then leave "Note for future 12.x.x releases:" unchanged

Reviewer 2: @sslavic thanks that makes better sense. But should change "AWS v12.2.x" to "KVM v12.1.x+" right

Reviewer 3: Yes, AWS reference is wrong, it's KVM release

Author: Can I commit what @cokiengchiara suggest earlier in this thread?

Reviewer 2: thanks, I was asking to confirm if it should be 12.1, not 12.2 :D but I think that's right, so will commit this.

## Change details

### app-operator [v2.3.3](https://github.com/giantswarm/app-operator/blob/master/CHANGELOG.md#233---2020-10-15)
### chart-operator [v2.3.5](https://github.com/giantswarm/chart-operator/blob/master/CHANGELOG.md#233---2020-09-29)
- Updated Helm to [v3.3.4](https://github.com/helm/helm/releases/tag/v3.3.4).

### kvm-operator [v3.12.2](https://github.com/giantswarm/kvm-operator/blob/master/CHANGELOG.md#3122---2020-10-16)
- Added monitoring labels into prometheus metrics

> ### containerlinux [2512.5.0](https://www.flatcar-linux.org/releases/#release-2512.5.0)

Reviewer 2: ```suggestion
### Container Linux [2512.5.0](https://www.flatcar-linux.org/releases/#release-2512.5.0)
```

## :zap:  Giant Swarm Release v12.3.0 for KVM :zap:

This release upgrades all Helm releases managed by Giant Swarm to use [Helm v3.3.4](https://github.com/helm/helm/releases/tag/v3.3.4).

This lets us benefit from the improved security model and keep up to date with the community. We also remove the Tiller deployment from the giantswarm namespace, removing its gRPC endpoint, which reduces operational complexity.

If you are still using Helm 2 then these Helm releases will not be affected. However we encourage you to upgrade to Helm 3. As Helm 2 support ends on November 13th 2020. https://helm.sh/blog/helm-v2-deprecation-timeline/

Below, you can find more details on components that were changed with this release.

**Note before upgrade:**

Please contact your Solution Engineer before upgrading. The upgrade is automated. However, it includes a data migration from Helm 2 release configmaps to Helm 3 release secrets, there are some pre-upgrade checks and we recommend monitoring the upgrade to ensure safety.

**Note for Solution Engineers:**

> Please ensure cluster is on [KVM 12.1.x](https://github.com/giantswarm/releases/tree/master/kvm/v12.1.0) platform release first before upgrading to 12.2.0+

Reviewer 2: ```suggestion
Before upgrading, please ensure cluster is on [KVM 12.1.x](https://github.com/giantswarm/releases/tree/master/kvm/v12.1.0) platform release first.
```

## :zap:  Giant Swarm Release v12.3.0 for KVM :zap:

This release upgrades all Helm releases managed by Giant Swarm to use [Helm v3.3.4](https://github.com/helm/helm/releases/tag/v3.3.4).

This lets us benefit from the improved security model and keep up to date with the community. We also remove the Tiller deployment from the giantswarm namespace, removing its gRPC endpoint, which reduces operational complexity.

> If you are still using Helm 2 then these Helm releases will not be affected. However we encourage you to upgrade to Helm 3. As Helm 2 support ends on November 13th 2020. https://helm.sh/blog/helm-v2-deprecation-timeline/

Reviewer 4: ```suggestion
If you are still using Helm 2 then these Helm releases will not be affected. However we encourage you to upgrade to Helm 3 as Helm 2 support ends on November 13th 2020. https://helm.sh/blog/helm-v2-deprecation-timeline/
```
